// plugin-welcome/src/api/models/EntEmployeeEdges.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Example API
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import {
    EntReturninvoice,
    EntReturninvoiceFromJSON,
    EntReturninvoiceFromJSONTyped,
    EntReturninvoiceToJSON,
} from './';

/**
 * 
 * @export
 * @interface EntEmployeeEdges
 */
export interface EntEmployeeEdges {
    /**
     * Employees holds the value of the employees edge.
     * @type {Array<EntReturninvoice>}
     * @memberof EntEmployeeEdges
     */
    employees?: Array<EntReturninvoice>;
}

export function EntEmployeeEdgesFromJSON(json: any): EntEmployeeEdges {
    return EntEmployeeEdgesFromJSONTyped(json, false);
}

export function EntEmployeeEdgesFromJSONTyped(json: any, ignoreDiscriminator: boolean): EntEmployeeEdges {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'employees': !exists(json, 'employees') ? undefined : ((json['employees'] as Array<any>).map(EntReturninvoiceFromJSON)),
    };
}

export function EntEmployeeEdgesToJSON(value?: EntEmployeeEdges | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'employees': value.employees === undefined ? undefined : ((value.employees as Array<any>).map(EntReturninvoiceToJSON)),
    };
}
