// plugin-welcome/src/api/apis/DefaultApi.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Example API
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */


import * as runtime from '../runtime';
import {
    ControllersReturninvoice,
    ControllersReturninvoiceFromJSON,
    ControllersReturninvoiceToJSON,
    EntEmployee,
    EntEmployeeFromJSON,
    EntEmployeeToJSON,
    EntRepairinvoice,
    EntRepairinvoiceFromJSON,
    EntRepairinvoiceToJSON,
    EntReturninvoice,
    EntReturninvoiceFromJSON,
    EntReturninvoiceToJSON,
    EntStatust,
    EntStatustFromJSON,
    EntStatustToJSON,
} from '../models';

export interface CreateEmployeeRequest {
    employee: EntEmployee;
}

export interface CreateRepairinvoiceRequest {
    repairinvoice: EntRepairinvoice;
}

export interface CreateReturninvoiceRequest {
    returninvoice: ControllersReturninvoice;
}

export interface CreateStatustRequest {
    statust: EntStatust;
}

export interface DeleteEmployeeRequest {
    id: number;
}

export interface DeleteRepairinvoiceRequest {
    id: number;
}

export interface DeleteStatustRequest {
    id: number;
}

export interface GetEmployeeRequest {
    id: number;
}

export interface GetRepairinvoiceRequest {
    id: number;
}

export interface GetStatustRequest {
    id: number;
}

export interface ListEmployeeRequest {
    limit?: number;
    offset?: number;
}

export interface ListRepairinvoiceRequest {
    limit?: number;
    offset?: number;
}

export interface ListReturninvoiceRequest {
    limit?: number;
    offset?: number;
}

export interface ListStatustRequest {
    limit?: number;
    offset?: number;
}

/**
 * 
 */
export class DefaultApi extends runtime.BaseAPI {

    /**
     * Create employee
     * Create employee
     */
    async createEmployeeRaw(requestParameters: CreateEmployeeRequest): Promise<runtime.ApiResponse<EntEmployee>> {
        if (requestParameters.employee === null || requestParameters.employee === undefined) {
            throw new runtime.RequiredError('employee','Required parameter requestParameters.employee was null or undefined when calling createEmployee.');
        }

        const queryParameters: runtime.HTTPQuery = {};

        const headerParameters: runtime.HTTPHeaders = {};

        headerParameters['Content-Type'] = 'application/json';

        const response = await this.request({
            path: `/employees`,
            method: 'POST',
            headers: headerParameters,
            query: queryParameters,
            body: EntEmployeeToJSON(requestParameters.employee),
        });

        return new runtime.JSONApiResponse(response, (jsonValue) => EntEmployeeFromJSON(jsonValue));
    }

    /**
     * Create employee
     * Create employee
     */
    async createEmployee(requestParameters: CreateEmployeeRequest): Promise<EntEmployee> {
        const response = await this.createEmployeeRaw(requestParameters);
        return await response.value();
    }

    /**
     * Create repairinvoice
     * Create repairinvoice
     */
    async createRepairinvoiceRaw(requestParameters: CreateRepairinvoiceRequest): Promise<runtime.ApiResponse<EntRepairinvoice>> {
        if (requestParameters.repairinvoice === null || requestParameters.repairinvoice === undefined) {
            throw new runtime.RequiredError('repairinvoice','Required parameter requestParameters.repairinvoice was null or undefined when calling createRepairinvoice.');
        }

        const queryParameters: runtime.HTTPQuery = {};

        const headerParameters: runtime.HTTPHeaders = {};

        headerParameters['Content-Type'] = 'application/json';

        const response = await this.request({
            path: `/repairinvoices`,
            method: 'POST',
            headers: headerParameters,
            query: queryParameters,
            body: EntRepairinvoiceToJSON(requestParameters.repairinvoice),
        });

        return new runtime.JSONApiResponse(response, (jsonValue) => EntRepairinvoiceFromJSON(jsonValue));
    }

    /**
     * Create repairinvoice
     * Create repairinvoice
     */
    async createRepairinvoice(requestParameters: CreateRepairinvoiceRequest): Promise<EntRepairinvoice> {
        const response = await this.createRepairinvoiceRaw(requestParameters);
        return await response.value();
    }

    /**
     * Create returninvoice
     * Create returninvoice
     */
    async createReturninvoiceRaw(requestParameters: CreateReturninvoiceRequest): Promise<runtime.ApiResponse<ControllersReturninvoice>> {
        if (requestParameters.returninvoice === null || requestParameters.returninvoice === undefined) {
            throw new runtime.RequiredError('returninvoice','Required parameter requestParameters.returninvoice was null or undefined when calling createReturninvoice.');
        }

        const queryParameters: runtime.HTTPQuery = {};

        const headerParameters: runtime.HTTPHeaders = {};

        headerParameters['Content-Type'] = 'application/json';

        const response = await this.request({
            path: `/returninvoices`,
            method: 'POST',
            headers: headerParameters,
            query: queryParameters,
            body: ControllersReturninvoiceToJSON(requestParameters.returninvoice),
        });

        return new runtime.JSONApiResponse(response, (jsonValue) => ControllersReturninvoiceFromJSON(jsonValue));
    }

    /**
     * Create returninvoice
     * Create returninvoice
     */
    async createReturninvoice(requestParameters: CreateReturninvoiceRequest): Promise<ControllersReturninvoice> {
        const response = await this.createReturninvoiceRaw(requestParameters);
        return await response.value();
    }

    /**
     * Create statust
     * Create statust
     */
    async createStatustRaw(requestParameters: CreateStatustRequest): Promise<runtime.ApiResponse<EntStatust>> {
        if (requestParameters.statust === null || requestParameters.statust === undefined) {
            throw new runtime.RequiredError('statust','Required parameter requestParameters.statust was null or undefined when calling createStatust.');
        }

        const queryParameters: runtime.HTTPQuery = {};

        const headerParameters: runtime.HTTPHeaders = {};

        headerParameters['Content-Type'] = 'application/json';

        const response = await this.request({
            path: `/statusts`,
            method: 'POST',
            headers: headerParameters,
            query: queryParameters,
            body: EntStatustToJSON(requestParameters.statust),
        });

        return new runtime.JSONApiResponse(response, (jsonValue) => EntStatustFromJSON(jsonValue));
    }

    /**
     * Create statust
     * Create statust
     */
    async createStatust(requestParameters: CreateStatustRequest): Promise<EntStatust> {
        const response = await this.createStatustRaw(requestParameters);
        return await response.value();
    }

    /**
     * get employee by ID
     * Delete a employee entity by ID
     */
    async deleteEmployeeRaw(requestParameters: DeleteEmployeeRequest): Promise<runtime.ApiResponse<object>> {
        if (requestParameters.id === null || requestParameters.id === undefined) {
            throw new runtime.RequiredError('id','Required parameter requestParameters.id was null or undefined when calling deleteEmployee.');
        }

        const queryParameters: runtime.HTTPQuery = {};

        const headerParameters: runtime.HTTPHeaders = {};

        const response = await this.request({
            path: `/employee/{id}`.replace(`{${"id"}}`, encodeURIComponent(String(requestParameters.id))),
            method: 'DELETE',
            headers: headerParameters,
            query: queryParameters,
        });

        return new runtime.JSONApiResponse<any>(response);
    }

    /**
     * get employee by ID
     * Delete a employee entity by ID
     */
    async deleteEmployee(requestParameters: DeleteEmployeeRequest): Promise<object> {
        const response = await this.deleteEmployeeRaw(requestParameters);
        return await response.value();
    }

    /**
     * get repairinvoice by ID
     * Delete a repairinvoice entity by ID
     */
    async deleteRepairinvoiceRaw(requestParameters: DeleteRepairinvoiceRequest): Promise<runtime.ApiResponse<object>> {
        if (requestParameters.id === null || requestParameters.id === undefined) {
            throw new runtime.RequiredError('id','Required parameter requestParameters.id was null or undefined when calling deleteRepairinvoice.');
        }

        const queryParameters: runtime.HTTPQuery = {};

        const headerParameters: runtime.HTTPHeaders = {};

        const response = await this.request({
            path: `/repairinvoice/{id}`.replace(`{${"id"}}`, encodeURIComponent(String(requestParameters.id))),
            method: 'DELETE',
            headers: headerParameters,
            query: queryParameters,
        });

        return new runtime.JSONApiResponse<any>(response);
    }

    /**
     * get repairinvoice by ID
     * Delete a repairinvoice entity by ID
     */
    async deleteRepairinvoice(requestParameters: DeleteRepairinvoiceRequest): Promise<object> {
        const response = await this.deleteRepairinvoiceRaw(requestParameters);
        return await response.value();
    }

    /**
     * get statust by ID
     * Delete a statust entity by ID
     */
    async deleteStatustRaw(requestParameters: DeleteStatustRequest): Promise<runtime.ApiResponse<object>> {
        if (requestParameters.id === null || requestParameters.id === undefined) {
            throw new runtime.RequiredError('id','Required parameter requestParameters.id was null or undefined when calling deleteStatust.');
        }

        const queryParameters: runtime.HTTPQuery = {};

        const headerParameters: runtime.HTTPHeaders = {};

        const response = await this.request({
            path: `/statust/{id}`.replace(`{${"id"}}`, encodeURIComponent(String(requestParameters.id))),
            method: 'DELETE',
            headers: headerParameters,
            query: queryParameters,
        });

        return new runtime.JSONApiResponse<any>(response);
    }

    /**
     * get statust by ID
     * Delete a statust entity by ID
     */
    async deleteStatust(requestParameters: DeleteStatustRequest): Promise<object> {
        const response = await this.deleteStatustRaw(requestParameters);
        return await response.value();
    }

    /**
     * get employee by ID
     * Get a employee entity by ID
     */
    async getEmployeeRaw(requestParameters: GetEmployeeRequest): Promise<runtime.ApiResponse<EntEmployee>> {
        if (requestParameters.id === null || requestParameters.id === undefined) {
            throw new runtime.RequiredError('id','Required parameter requestParameters.id was null or undefined when calling getEmployee.');
        }

        const queryParameters: runtime.HTTPQuery = {};

        const headerParameters: runtime.HTTPHeaders = {};

        const response = await this.request({
            path: `/employees/{id}`.replace(`{${"id"}}`, encodeURIComponent(String(requestParameters.id))),
            method: 'GET',
            headers: headerParameters,
            query: queryParameters,
        });

        return new runtime.JSONApiResponse(response, (jsonValue) => EntEmployeeFromJSON(jsonValue));
    }

    /**
     * get employee by ID
     * Get a employee entity by ID
     */
    async getEmployee(requestParameters: GetEmployeeRequest): Promise<EntEmployee> {
        const response = await this.getEmployeeRaw(requestParameters);
        return await response.value();
    }

    /**
     * get repairinvoice by ID
     * Get a repairinvoice entity by ID
     */
    async getRepairinvoiceRaw(requestParameters: GetRepairinvoiceRequest): Promise<runtime.ApiResponse<EntRepairinvoice>> {
        if (requestParameters.id === null || requestParameters.id === undefined) {
            throw new runtime.RequiredError('id','Required parameter requestParameters.id was null or undefined when calling getRepairinvoice.');
        }

        const queryParameters: runtime.HTTPQuery = {};

        const headerParameters: runtime.HTTPHeaders = {};

        const response = await this.request({
            path: `/repairinvoices/{id}`.replace(`{${"id"}}`, encodeURIComponent(String(requestParameters.id))),
            method: 'GET',
            headers: headerParameters,
            query: queryParameters,
        });

        return new runtime.JSONApiResponse(response, (jsonValue) => EntRepairinvoiceFromJSON(jsonValue));
    }

    /**
     * get repairinvoice by ID
     * Get a repairinvoice entity by ID
     */
    async getRepairinvoice(requestParameters: GetRepairinvoiceRequest): Promise<EntRepairinvoice> {
        const response = await this.getRepairinvoiceRaw(requestParameters);
        return await response.value();
    }

    /**
     * get statust by ID
     * Get a statust entity by ID
     */
    async getStatustRaw(requestParameters: GetStatustRequest): Promise<runtime.ApiResponse<EntStatust>> {
        if (requestParameters.id === null || requestParameters.id === undefined) {
            throw new runtime.RequiredError('id','Required parameter requestParameters.id was null or undefined when calling getStatust.');
        }

        const queryParameters: runtime.HTTPQuery = {};

        const headerParameters: runtime.HTTPHeaders = {};

        const response = await this.request({
            path: `/statusts/{id}`.replace(`{${"id"}}`, encodeURIComponent(String(requestParameters.id))),
            method: 'GET',
            headers: headerParameters,
            query: queryParameters,
        });

        return new runtime.JSONApiResponse(response, (jsonValue) => EntStatustFromJSON(jsonValue));
    }

    /**
     * get statust by ID
     * Get a statust entity by ID
     */
    async getStatust(requestParameters: GetStatustRequest): Promise<EntStatust> {
        const response = await this.getStatustRaw(requestParameters);
        return await response.value();
    }

    /**
     * list employee entities
     * List employee entities
     */
    async listEmployeeRaw(requestParameters: ListEmployeeRequest): Promise<runtime.ApiResponse<Array<EntEmployee>>> {
        const queryParameters: runtime.HTTPQuery = {};

        if (requestParameters.limit !== undefined) {
            queryParameters['limit'] = requestParameters.limit;
        }

        if (requestParameters.offset !== undefined) {
            queryParameters['offset'] = requestParameters.offset;
        }

        const headerParameters: runtime.HTTPHeaders = {};

        const response = await this.request({
            path: `/employees`,
            method: 'GET',
            headers: headerParameters,
            query: queryParameters,
        });

        return new runtime.JSONApiResponse(response, (jsonValue) => jsonValue.map(EntEmployeeFromJSON));
    }

    /**
     * list employee entities
     * List employee entities
     */
    async listEmployee(requestParameters: ListEmployeeRequest): Promise<Array<EntEmployee>> {
        const response = await this.listEmployeeRaw(requestParameters);
        return await response.value();
    }

    /**
     * list repairinvoice entities
     * List repairinvoice entities
     */
    async listRepairinvoiceRaw(requestParameters: ListRepairinvoiceRequest): Promise<runtime.ApiResponse<Array<EntRepairinvoice>>> {
        const queryParameters: runtime.HTTPQuery = {};

        if (requestParameters.limit !== undefined) {
            queryParameters['limit'] = requestParameters.limit;
        }

        if (requestParameters.offset !== undefined) {
            queryParameters['offset'] = requestParameters.offset;
        }

        const headerParameters: runtime.HTTPHeaders = {};

        const response = await this.request({
            path: `/repairinvoices`,
            method: 'GET',
            headers: headerParameters,
            query: queryParameters,
        });

        return new runtime.JSONApiResponse(response, (jsonValue) => jsonValue.map(EntRepairinvoiceFromJSON));
    }

    /**
     * list repairinvoice entities
     * List repairinvoice entities
     */
    async listRepairinvoice(requestParameters: ListRepairinvoiceRequest): Promise<Array<EntRepairinvoice>> {
        const response = await this.listRepairinvoiceRaw(requestParameters);
        return await response.value();
    }

    /**
     * list returninvoice entities
     * List returninvoice entities
     */
    async listReturninvoiceRaw(requestParameters: ListReturninvoiceRequest): Promise<runtime.ApiResponse<Array<EntReturninvoice>>> {
        const queryParameters: runtime.HTTPQuery = {};

        if (requestParameters.limit !== undefined) {
            queryParameters['limit'] = requestParameters.limit;
        }

        if (requestParameters.offset !== undefined) {
            queryParameters['offset'] = requestParameters.offset;
        }

        const headerParameters: runtime.HTTPHeaders = {};

        const response = await this.request({
            path: `/returninvoices`,
            method: 'GET',
            headers: headerParameters,
            query: queryParameters,
        });

        return new runtime.JSONApiResponse(response, (jsonValue) => jsonValue.map(EntReturninvoiceFromJSON));
    }

    /**
     * list returninvoice entities
     * List returninvoice entities
     */
    async listReturninvoice(requestParameters: ListReturninvoiceRequest): Promise<Array<EntReturninvoice>> {
        const response = await this.listReturninvoiceRaw(requestParameters);
        return await response.value();
    }

    /**
     * list statust entities
     * List statust entities
     */
    async listStatustRaw(requestParameters: ListStatustRequest): Promise<runtime.ApiResponse<Array<EntStatust>>> {
        const queryParameters: runtime.HTTPQuery = {};

        if (requestParameters.limit !== undefined) {
            queryParameters['limit'] = requestParameters.limit;
        }

        if (requestParameters.offset !== undefined) {
            queryParameters['offset'] = requestParameters.offset;
        }

        const headerParameters: runtime.HTTPHeaders = {};

        const response = await this.request({
            path: `/statusts`,
            method: 'GET',
            headers: headerParameters,
            query: queryParameters,
        });

        return new runtime.JSONApiResponse(response, (jsonValue) => jsonValue.map(EntStatustFromJSON));
    }

    /**
     * list statust entities
     * List statust entities
     */
    async listStatust(requestParameters: ListStatustRequest): Promise<Array<EntStatust>> {
        const response = await this.listStatustRaw(requestParameters);
        return await response.value();
    }

}
